python/pyarrow/io.pxi

```diff
@@ -1446,6 +1446,41 @@ cdef class Buffer(_Weakrefable):
         """
         return _wrap_device_allocation_type(self.buffer.get().device_type())
 
+    def copy(self, destination):
+        """
+        Copy the buffer to the destination device.
+
+        The buffer contents will be copied into a new buffer allocated onto
+        the destination MemoryManager or Device. It will return the new Buffer.
+        This function supports cross-device copies.
+
+        Parameters
+        ----------
+        destination : pyarrow.MemoryManager or pyarrow.Device
```

Comment:
Not required, but it may be nice to also allow a MemoryPool here.

Reply:
We don't have a way to map a given MemoryPool to a device manager (AFAIK), so in that case you mean to use a different API under the hood to copy to the memory pool? (`Buffer::CopySlice` seems to allow that, or first explicitly allocate a new Buffer of the correct size using the memory pool and then copying the data to it?)

Reply:
You can use https://github.com/apache/arrow/blob/c557fe51b3763b9492392f48c5ebcae5a1dd0b42/cpp/src/arrow/device.h#L295-L298